Remotion/Data/DomainObjects.UnitTests/Queries/ConfigurationLoader/BaseDirectoryBasedQueryFileFinderTest.cs

```diff
@@ -0,0 +1,129 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.IO;
+using JetBrains.Annotations;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Queries.ConfigurationLoader;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Queries.ConfigurationLoader
+{
+  [TestFixture]
+  public class BaseDirectoryBasedQueryFileFinderTest
+  {
+    private class QueryFileFinderStub : BaseDirectoryBasedQueryFileFinder
+    {
+      public QueryFileFinderStub ([NotNull] IAppContextProvider appContextProvider, [NotNull] string queryFile)
+          : base(appContextProvider, queryFile)
+      {
+      }
+
+      public QueryFileFinderStub ([NotNull] IAppContextProvider appContextProvider, [NotNull] string queryFile, bool optional)
+          : base(appContextProvider, queryFile, optional)
+      {
+      }
+    }
+
+    [Test]
+    public void GetQueryFilePaths_WithNonExistentFullPath_ThrowsException ()
+    {
+      var appContextProviderMock = new Mock<IAppContextProvider>();
+      appContextProviderMock.SetupGet(e => e.BaseDirectory).Returns(@"C:\base");
+
+      var queryFileFinder = new QueryFileFinderStub(appContextProviderMock.Object, @"C:\fullpath\query.txt");
+      Assert.That(
+          () => queryFileFinder.GetQueryFilePaths(),
+          Throws.TypeOf<ConfigurationException>()
+              .With.Message.EqualTo(@"The query file 'C:\fullpath\query.txt' does not exist."));
+    }
+
+    [Test]
+    public void GetQueryFilePaths_WithNonExistentFullPathButOptional_ReturnsNothing ()
+    {
+      var appContextProviderMock = new Mock<IAppContextProvider>();
+      appContextProviderMock.SetupGet(e => e.BaseDirectory).Returns(@"C:\base");
+
+      var queryFileFinder = new QueryFileFinderStub(appContextProviderMock.Object, @"C:\fullpath\query.txt", true);
+      Assert.That(queryFileFinder.GetQueryFilePaths(), Is.Empty);
+    }
```

Comment:
The difference between these two tests is that the query file is required in the first but optional in the second - this distinction is not that obvious in the names. Also, the distinction between these two tests and those for relative paths should be made more obvious.
Suggestion:
GetQueryFilePaths_FullPath_RequiredFile_NotFound_ThrowsException
GetQueryFilePaths_FullPath_OptionalFile_NotFound_ReturnsNothing (or "Empty")